ecs-init/docker/docker.go

```diff
@@ -26,13 +26,32 @@ import (
 )
 
 const (
-	logDir      = "/log"
-	dataDir     = "/data"
-	readOnly    = ":ro"
+	// logDir specifies the location of Agent log files in the container
+	logDir = "/log"
+	// dataDir specifies the location of Agent state file in the container
+	dataDir = "/data"
+	// readOnly specifies the read-only suffix for mounting host volumes
+	// when creating the Agent container
+	readOnly = ":ro"
+	// hostProcDir binds the host's /proc directory to /host/proc within the
+	// ECS Agent container
+	// The ECS Agent needs access to host's /proc directory when configuring
+	// the network namespace of containers for tasks that are configured
+	// with an ENI
 	hostProcDir = "/host/proc"
-	// set default to /var/run instead of /var/run/docker.sock in case
-	// /var/run/docker.sock is deleted and recreated outside the container
+	// defaultDockerEndpoint is set to /var/run instead of /var/run/docker.sock
+	// in case /var/run/docker.sock is deleted and recreated outside the container
 	defaultDockerEndpoint = "/var/run"
+	// dhclientLeasesLocation specifies the location where dhclient leases
+	// information is tracked in the Agent container
+	dhclientLeasesLocation = "/var/lib/dhclient"
+	// dhclientLibDir specifies the location of shared libraries on the
```

Comment:
Is there a dependency on a specific version of libc here?  Are you mounting in the dhclient executable from the host?

Reply:
We're mounting `libc` from the host.

Reply:
That location does not exists in SLES. SLES uses wicked for networking and leases stored in a different location in XML format, for example:

# cat /var/lib/wicked/lease-eth0-dhcp-ipv4.xml
<lease>
  <family>ipv4</family>
  <type>dhcp</type>
  <uuid>b3bda659-d649-0800-e403-000004000000</uuid>
  <state>granted</state>
  <update>0x00000000</update>
  <acquired>1504272564</acquired>
  <ipv4:dhcp>
    <client-id>01:42:01:0a:f0:00:06</client-id>
    <server-id>169.254.169.254</server-id>
    <relay-address>10.240.0.1</relay-address>
    <sender-hw-address>42:01:0a:f0:00:01</sender-hw-address>
    <lease-time>86400</lease-time>
    <renewal-time>43200</renewal-time>
    <rebind-time>75600</rebind-time>
    <hostname>sles12devtestguest.c.suse-gce-test.internal</hostname>
    <address>10.240.0.6</address>
    <netmask>255.255.255.255</netmask>
    <mtu>1460</mtu>
    <boot>
      <server-address>10.240.0.1</server-address>
    </boot>
    <routes>
      <route>
        <destination>10.240.0.1/32</destination>
      </route>
      <route>
        <nexthop>
          <gateway>10.240.0.1</gateway>
        </nexthop>
      </route>
    </routes>
    <dns>
      <domain>c.suse-gce-test.internal</domain>
      <server>169.254.169.254</server>
      <search>c.suse-gce-test.internal</search>
      <search>google.internal</search>
    </dns>
    <ntp>
      <server>169.254.169.254</server>
    </ntp>
  </ipv4:dhcp>
</lease>

Reply:
@rjschwei can you please let me know which location are you referring to? If you're referring to `dhclientLeasesLocation = "/var/lib/dhclient"`, that's something that need not be present on the host. It's a volume mount target within the Agent container.

Reply:
@aaithal well dhclientLeasesLocation = "/var/lib/dhclient" needs to be present now, but it is not present on SLES and will not be, basically this update breaks SUSE support.

Leas information is located in /var/lib/wicked in SLES and the file is in a different format than the dhclient lease file

Reply:
@rjschwei It seems there's a bit of disconnect here and I apologize for that. Let me try to provide some details here. The `dhclientLeasesLocation = "/var/lib/dhclient"` specifies the location where the ECS Agent will write lease information to. Since the ECS Agent will be launched as a container, it can have it's own filesystem structure that's different than that of the host. So, the ECS Agent container will write leases file in this location. The way we're doing this for Amazon Linux is by creating the `"/var/lib/ecs/dhclient"` on the host (you can find a refernece to this here: https://github.com/aws/amazon-ecs-init/blob/enis/ecs-init/config/common.go#L81) via the `ecs-init.spec` file in https://github.com/aws/amazon-ecs-init/blob/enis/packaging/amazon-linux-ami/ecs-init.spec#L79.

Essentially, ECS Agent will be launched with `-v /var/lib/ecs/dhclient:/var/lib/dhclient`. The host part of this mount will be created via the RPM spec file. I just had a brief look at the spec file for suse and it seems that the install section can be modified to create this on suse as well: https://github.com/aws/amazon-ecs-init/blob/enis/packaging/suse/amazon-ecs-init.spec#L48 (please correct me if i'm wrong).

Reply:
Well, yes we can create any path via a spec file thus on the host we can certainly create /var/lib/dhclient and if that information is not actually needed to be shared with the host than there is no problem. However, if the host for some reason or another needs to be aware of the lease information then that is not going to work on SLES.

Further, if the agent moves into a container from running directly on the host then somehow that container has to be pulled onto the host, and I obviously do not follow the work here  closely enough to know how that is being resolved. Anyway, if the agent runs in a container and you provide the container then of course any dhcp handler of choice can be used. However, if the expectation is that I build a container that contains the agent and then build that container into the VM image and launch it on start up then the use of dhclient is a problem.